### PR TITLE
build: migrate to dart 3

### PR DESCRIPTION
**Pull request by 2shrestha22**
This PR updates minimum dart version to 3.0.0.

### CODE REVIEW

**Review by EvadeMaster — APPROVED** [review]
👍 Looks fine to me!

File: pubspec.yaml [hunk 1]
```diff
@@ -74,7 +74,7 @@ dependencies:
   wakelock: ^0.6.2
   flutter_dotenv: ^5.0.2
   flutter_markdown: ^0.6.14
-  pub_release: ^8.0.3
+  # pub_release: ^8.0.3
```
Comment (Ushie) [inline pubspec.yaml:77]: Why was this commented out?

File: lib/ui/views/app_selector/app_selector_view.dart [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide SearchBar;
```
Comment (Ushie) [inline lib/ui/views/app_selector/app_selector_view.dart:1]: Why is SearchBar being hidden?

File: lib/ui/views/patches_selector/patches_selector_view.dart [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide SearchBar;
```
Comment (Ushie) [inline lib/ui/views/patches_selector/patches_selector_view.dart:1]: Why is SearchBar being hidden?

**Review by Ushie — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (2shrestha22) [reply pubspec.yaml:77]: It was not needed as app dependency.

File: lib/ui/views/app_selector/app_selector_view.dart (hunk 2 above)
Comment (2shrestha22) [reply lib/ui/views/app_selector/app_selector_view.dart:1]: Newer Flutter version have SearchBar which conflicts with already using SearchBar.

File: pubspec.yaml (hunk 1 above)
Comment (2shrestha22) [reply pubspec.yaml]: I have removed the commented part.

Comment (Ushie) [reply pubspec.yaml]: Sorry, I meant remove the dependency entirely

Comment (Ushie) [reply pubspec.yaml]: Thanks!

**Review by Aunali321 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**